### DISCUSSION TITLE
May 2022 SP-API Release Announcement #2289 And Update our library

### DISCUSSION THREAD

**Discussion by abuzuhri** [abuzuhri/Amazon-SP-API-CSharp#191] [Announcements]
# Updated in our library for Amazon May 2022 SP-API Release

---
## New version of Catalog Items API

- [x] The new [Catalog Items API version 2022-04-01](https://developer-docs.amazon.com/sp-api/docs/catalog-items-api-v2022-04-01-reference) improves the API experience and addresses developer feedback with the following changes:

- [x] Added identifier-based searches to the searchCatalogItems operation, enabling batch-like requests with up to 20 identifiers per request (ASINs, UPCs, EANs, etc.)
- [x] Added dimensions data set, providing normalized item and package dimensions
- [x] Added display names for product category, product subcategory, and browse classification
- [x] Expanded identifying attributes to item summaries (item classification, package quantity, part number, website display group)
- [x] Added browse classifications and website display groups to sales ranks
- [x] Aligned searchCatalogItems and getCatalogItem operations to use the same data sets, including attributes
- [x] Replaced variations with relationships to allow for multiple relationship types to be represented, such as package hierarchies
- [x] Added variation theme details (name to the relationships details)

---
## New batch operations capabilities

- [x] The [Product Fees API](https://developer-docs.amazon.com/sp-api/docs/product-fees-api-v0-reference) and [Product Pricing API](https://developer-docs.amazon.com/sp-api/docs/product-pricing-api-v0-reference) now include batch operations for retrieving information for a list of products. We strongly encourage developers migrating to these APIs to use the new batch operations to most efficiently call these APIs instead of calling these APIs one-by-one.

- [x] [Product Fees API](https://developer-docs.amazon.com/sp-api/docs/product-fees-api-v0-reference) includes a new getMyFeesEstimates operation that retrieves product fee estimates for a list of products and marketplaces. For more information, see the [Product Fees API Use Case Guide](https://developer-docs.amazon.com/sp-api/docs/product-fees-v0-use-case-guide).
- [x] [Product Pricing API](https://developer-docs.amazon.com/sp-api/docs/product-pricing-api-v0-reference) includes new getItemOffersBatch and getListingOffersBatch operations that retrieves product pricing and offer information in a list for products. For more information, see the [Product Pricing API Use Case Guide](https://developer-docs.amazon.com/sp-api/docs/product-pricing-api-v0-use-case-guide).

---
## New Purchase Shield program operations

- [ ] The [Orders API](https://developer-docs.amazon.com/sp-api/docs/orders-api-v0-reference) now includes two new operations to support the Purchase Shield program for managing regulated orders in the UK and Italy. Regulated orders are a limited set of orders that fulfill products requiring a prescription (pet medication, prescription contacts, and prescription glasses). Two new operations have been added: getOrderRegulatedInfo and updateVerificationStatus.

---
## New Extended Producer Responsibility (EPR) report types

- [ ] The [Reports API](https://developer-docs.amazon.com/sp-api/docs/reports-api-v2021-06-30-reference) now includes three new [Extended Producer Responsibility (EPR) report types](https://developer-docs.amazon.com/sp-api/docs/report-type-values#regulatory-compliance-reports). The EPR Monthly Report, EPR Quarterly Report, and EPR Annual Reports include data on the EPR declarations for service providers in France and Germany to help sellers maintain compliance with EU regulations.

---
## New support for filtering and limiting Any Offer Changed notifications

- [x] The [Notifications API](https://developer-docs.amazon.com/sp-api/docs/notifications-api-v1-reference) ANY_OFFER_CHANGED notification now supports filtering notifications to only specific marketplaceIds, and limiting notifications to one every five or ten minute interval. The updated notification accepts a new processingDirective parameter that lets you pass additional information to the subscription to control the processing of notifications. For more information, refer to the [Notifications Use Case Guide](https://developer-docs.amazon.com/sp-api/docs/notifications-api-v1-use-case-guide#any_offer_changed).

---
## Report Deprecations

Listings Reports
The [following reports](https://developer-docs.amazon.com/sp-api/changelog/deprecation-of-get_merchant_listings_defect_data-and-listings_defect_report_xl) will be unavailable as of June 15, 2022:

_GET_MERCHANT_LISTINGS_DEFECT_DATA_
_LISTINGS_DEFECT_REPORT_XL_
These reports are replaced by _GET_MERCHANTS_LISTINGS_FYP_REPORT_. If you have an application that is affected by this change, you must update your application to use the _GET_MERCHANTS_LISTINGS_FYP_REPORT_ report by June 15, 2022 to avoid service disruption.

Vendor Retail Analytics Reports
The following reports are announcing deprecation and will be unavailable as of June 27, 2022.

GET_VENDOR_SALES_DIAGNOSTIC_REPORT
GET_VENDOR_DEMAND_FORECAST_REPORT
GET_VENDOR_INVENTORY_HEALTH_AND_PLANNING
Each of these reports has a new version or replacement report available. For more information on migrating to the new versions and replacement reports, refer to the [Vendor Retail Analytics Reports Migration Guide](https://developer-docs.amazon.com/sp-api/docs/vendor-retail-analytics-reports-migration-guide).

Inventory Reports
The [following reports](https://developer-docs.amazon.com/sp-api/changelog/march-2022-sp-api-release-announcement#report-deprecations) will be unavailable as of June 27, 2022:

GET_FBA_FULFILLMENT_INVENTORY_HEALTH_DATA
GET_EXCESS_INVENTORY_DATA
GET_FBA_INVENTORY_AGED_DATA
These reports are replaced by GET_FBA_INVENTORY_PLANNING_DATA. If you have an application that is affected by this change, you must update your application to use the GET_FBA_INVENTORY_PLANNING_DATA report by June 27, 2022 to avoid service disruption.

Brand Analytics Reports
The following reports will be unavailable as of June 30, 2022:

GET_BRAND_ANALYTICS_ITEM_COMPARISON_REPORT
GET_BRAND_ANALYTICS_ALTERNATE_PURCHASE_REPORT
There are no planned replacements for these reports. If you have an application that is affected by this change, be sure to ensure your applications no longer use the deprecated report types after June 30, 2022.

**Comment by BuleMr** [comment]
Hello, I found this type of report missing，Vendor retail analytics reports：reportType value: GET_VENDOR_SALES_REPORT

**Comment by ahelphenstine** [comment]
Also the financial report GET_DATE_RANGE_FINANCIAL_TRANSACTION_DATA is missing as well.

**Comment by abuzuhri** [comment]
Missing report added

**Comment by abuzuhri** [comment]
### Product GetCatalogItem Version 2022-04-01
```CSharp
var data = await amazonConnection.CatalogItem.GetCatalogItem202204Async(
    new Parameter.CatalogItems.ParameterGetCatalogItem
            {
                ASIN = "B00JK2YANC",
                includedData = new[] { IncludedData.attributes, 
                                       IncludedData.salesRanks,
                                       IncludedData.summaries, 
                                       IncludedData.productTypes, 
                                       IncludedData.relationships, 
                                       IncludedData.dimensions, 
                                       IncludedData.identifiers, 
                                       IncludedData.images }
            });
```

### Product SearchCatalogItems Version 2022-04-01
```CSharp
var data = await amazonConnection.CatalogItem.SearchCatalogItems202204Async(
    new Parameter.CatalogItems.ParameterSearchCatalogItems202204
            {
                keywords = new[] { "vitamin c" },
                includedData = new[] { IncludedData.attributes, 
                                       IncludedData.salesRanks,
                                       IncludedData.summaries, 
                                       IncludedData.productTypes, 
                                       IncludedData.relationships, 
                                       IncludedData.dimensions, 
                                       IncludedData.identifiers, 
                                       IncludedData.images }
            });
```
